internal/variants/variants.go

```diff
@@ -71,8 +81,29 @@ func (q *Query) ValidateInput() error {
 	if q.Allele == "" {
 		return errors.New("missing allele")
 	}
-	if q.Coord == nil {
-		return errors.New("missing coordinate")
+	if err := q.validateCoordinates(); err != nil {
+		return fmt.Errorf("validating coordinates: %v", err)
+	}
+	return nil
+}
+
+func (q *Query) validateCoordinates() error {
+	var precisePosition, imprecisePosition bool
+	if q.Start != nil && (q.End != nil || q.Allele != "") {
+		precisePosition = true
+	}
+	if q.StartMin != nil && q.StartMax != nil && q.EndMin != nil && q.EndMax != nil {
+		imprecisePosition = true
+	}
+
+	if precisePosition && imprecisePosition {
+		return errors.New("please query either precise or imprecise position")
+	}
+	if precisePosition || imprecisePosition {
+		return nil
+	}
+	if q.Start != nil || q.End != nil || q.StartMin != nil || q.StartMax != nil || q.EndMin != nil || q.EndMax != nil {
```

Comment:
What are you guarding against here?  The case where no position at all is specified?

Reply:
The case where some coordinate parameters are specified but not a valid combination of them. ex just  end and startMin are specified. I've refactored that to cover the case where there are extra parameters beside a valid combination